neo/Consensus/ConsensusService.cs

```diff
@@ -272,12 +289,17 @@ private void OnPrepareRequestReceived(ConsensusPayload payload, PrepareRequest m
         private void OnPrepareResponseReceived(ConsensusPayload payload, PrepareResponse message)
         {
             Log($"{nameof(OnPrepareResponseReceived)}: height={payload.BlockIndex} view={message.ViewNumber} index={payload.ValidatorIndex}");
-            if (context.State.HasFlag(ConsensusState.BlockSent)) return;
             if (context.Signatures[payload.ValidatorIndex] != null) return;
-            Block header = context.MakeHeader();
-            if (header == null || !Crypto.Default.VerifySignature(header.GetHashData(), message.Signature, context.Validators[payload.ValidatorIndex].EncodePoint(false))) return;
-            context.Signatures[payload.ValidatorIndex] = message.Signature;
-            CheckSignatures();
+            byte[] hashData = context.MakeHeader()?.GetHashData();
+            if (hashData == null)
+            {
+                context.Signatures[payload.ValidatorIndex] = message.Signature;
```

Comment:
@erikzhang, I am double checking here and just to be sure I am thinking why `message.Signature` is assigned here to be teh Signatures array.

Should not it be `context.Signatures[payload.ValidatorIndex] = null`?
If the hashData is null it means that the payload is kind of empty, right?